clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -524,6 +524,48 @@ MDNode *LoopInfo::createMetadata(
     LoopProperties.push_back(MDNode::get(Ctx, Vals));
   }
 
+  if (Attrs.SYCLLoopCoalesceEnable) {
+    LLVMContext &Ctx = Header->getContext();
+    Metadata *Vals[] = {MDString::get(Ctx, "llvm.loop.coalesce.enable")};
+    LoopProperties.push_back(MDNode::get(Ctx, Vals));
+  }
+
+  if (Attrs.SYCLLoopCoalesceNLevels > 0) {
+    LLVMContext &Ctx = Header->getContext();
+    Metadata *Vals[] = {
+        MDString::get(Ctx, "llvm.loop.coalesce.count"),
+        ConstantAsMetadata::get(ConstantInt::get(
+            llvm::Type::getInt32Ty(Ctx), Attrs.SYCLLoopCoalesceNLevels))};
+    LoopProperties.push_back(MDNode::get(Ctx, Vals));
+  }
+
+  if (Attrs.SYCLLoopPipeliningDisable) {
+    LLVMContext &Ctx = Header->getContext();
+    Metadata *Vals[] = {MDString::get(Ctx, "llvm.loop.intel.pipelining.enable"),
+                        ConstantAsMetadata::get(
+                            ConstantInt::get(llvm::Type::getInt32Ty(Ctx), 0))};
+    LoopProperties.push_back(MDNode::get(Ctx, Vals));
+  }
+
+  if (Attrs.SYCLMaxInterleavingEnable) {
```

Comment:
From doc:
> Parameter N is mandatory

therefore

```suggestion
  if (Attrs.YCLMaxInterleavingNInvocations >= 0) {
```

and remove SYCLMaxInterleavingEnable afterwards.

Reply:
With that saying, I acknowledge, that SYCLMaxInterleavingNInvocations shall be initialized as '-1' by default. But I think removal of unnecessary class filed worth ir.

Reply:
I prefer leaving this in the same approach in which max_concurrency attribute was implemented.

Reply:
Ok, then I'll try to clean it up later.